src/main/java/Duke.java

```diff
@@ -1,10 +1,127 @@
-public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+import sampletest.Task;
+import sampletest.Events;
+import sampletest.Deadlines;
+import java.util.Scanner;
+import java.util.ArrayList;
+/**
+ * CS2103 Individual Project
+ * author Wei Cheng
+ * The class for the ChatBot
+ */
+public class Duke  {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor to create an instance of Duke.
+     * @param filePath the String representation of the file location
+     */
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError(e);
+            tasks = new TaskList();
+        }
     }
-}
+
+    /**
+     * To run the conversation with the user.
+     */
+    public void run() {
+        Scanner sc = new Scanner(System.in);
+        String[] userInput = sc.nextLine().split(" ",2);
+        Parser parser = new Parser(userInput);
+        String command = parser.getCommand();
+        while (!command.equals("bye")) {
+            try {
+                ExceptionGenerator.checkInputLength(userInput);
+                ExceptionGenerator.checkInputAction(userInput);
+                switch (command) {
```

Comment:
It is more appropriate to express the command in terms of enums, instead of operating on raw strings since your valid command space is finite and small.